run_bin_test.go

```diff
@@ -538,6 +590,74 @@ func TestCoverageCollector_RunBinary(t *testing.T) {
 			errMessage:   "testing: warning: no tests to run\n",
 			wantExitCode: -1,
 		},
+		{
+			name: "succeed running binary with stdin pipe preCmdFunc",
+			args: args{
+				binPath:      "./test_bins/read_stdin.sh",
+				mainTestName: "",
+				env:          nil,
+				args:         nil,
+			},
+			fields: fields{
+				MergedCoverageFilename: "temp_coverage.out",
+				CollectCoverage:        false,
+			},
+			wantOutput:   "Hello from prefunc\n",
+			wantExitCode: 1,
+			wantErr: false,
+			cmdFuncs: []CoverageCollectorOption{stdinPipePreFunc()},
+		},
+		{
+			name: "fail running binary with error in preCmdFunc",
+			args: args{
+				binPath:      "./test_bins/read_stdin.sh",
+				mainTestName: "",
+				env:          nil,
+				args:         nil,
+			},
+			fields: fields{
+				MergedCoverageFilename: "temp_coverage.out",
+				CollectCoverage:        false,
+			},
+			errMessage: "oh no!",
+			wantExitCode: -1,
+			wantErr: true,
+			cmdFuncs: []CoverageCollectorOption{errPreCmdFunc()},
+		},
+		{
+			name: "fail running binary with error in postCmdFunc",
+			args: args{
+				binPath:      "./test_bins/read_stdin.sh",
+				mainTestName: "",
+				env:          nil,
+				args:         nil,
+			},
+			fields: fields{
+				MergedCoverageFilename: "temp_coverage.out",
+				CollectCoverage:        false,
+			},
+			errMessage: "oh no!",
+			wantExitCode: -1,
+			wantErr: true,
+			cmdFuncs: []CoverageCollectorOption{errPostCmdFunc()},
+		},
+		{
+			name: "succeed running binary with pre and post cmdFuncs",
+			args: args{
+				binPath:      "./set_covermode",
+				mainTestName: "TestRunMain",
+				env:          nil,
+				args:         nil,
+			},
+			fields: fields{
+				MergedCoverageFilename: "temp_coverage.out",
+				CollectCoverage:        true,
+			},
+			wantOutput: "Hello world\n",
+			wantExitCode: 1,
+			wantErr: false,
+			cmdFuncs: []CoverageCollectorOption{nilPreCmdFunc(), nilPostCmdFunc()},
```

Comment:
How are you testing that these actually executed? Maybe have them print something and check if it got printed?

Reply:
added new tests that check for output to a buffer (see TestPreExec / TestPostExec)